io-sim/src/Control/Monad/IOSim.hs

```diff
@@ -191,6 +195,9 @@ instance MonadSay (SimM s) where
   say msg = SimM $ \k -> Say msg (k ())
 
 instance MonadFork (SimM s) where
+  type ThreadIdT (SimM s) = ThreadId
```

Comment:
I think we'll be able to call this `ThreadId` rather than `ThreadIdT`. See how we resolve the name clashes with `TVars` etc.